src/analysis/normed_space/inner_product.lean

```diff
@@ -0,0 +1,1553 @@
+lemma abs_inner_le_norm (x y : F) : abs ⟪x, y⟫ ≤ ∥x∥ * ∥y∥ :=
+nonneg_le_nonneg_of_squares_le (mul_nonneg (sqrt_nonneg _) (sqrt_nonneg _))
+begin
+  have H : ∥x∥ * ∥y∥ * (∥x∥ * ∥y∥) = re ⟪y, y⟫ * re ⟪x, x⟫,
+  { simp only [inner_self_eq_norm_square], ring, },
+  rw H,
+  conv
+  begin
+    to_lhs, congr, rw[inner_abs_conj_sym],
+  end,
+  exact inner_mul_inner_self_le y x,
+end
+
+/-- Normed group structure constructed from an `inner_product_space.core` structure -/
+def to_normed_group : normed_group F :=
+normed_group.of_core F
+{ norm_eq_zero_iff := assume x,
+  begin
+    split,
+    { intro H,
+      change sqrt (re ⟪x, x⟫) = 0 at H,
+      rw [sqrt_eq_zero inner_self_nonneg] at H,
+      apply (inner_self_eq_zero : ⟪x, x⟫ = 0 ↔ x = 0).mp,
+      rw ext_iff,
+      exact ⟨by simp [H], by simp [inner_self_im_zero]⟩ },
+    { rintro rfl,
+      change sqrt (re ⟪0, 0⟫) = 0,
+      simp only [sqrt_zero, inner_zero_right, add_monoid_hom.map_zero] }
+  end,
+  triangle := assume x y,
+  begin
+    have h₁ : abs ⟪x, y⟫ ≤ ∥x∥ * ∥y∥ := abs_inner_le_norm _ _,
+    have h₂ : re ⟪x, y⟫ ≤ abs ⟪x, y⟫ := re_le_abs _,
+    have h₃ : re ⟪x, y⟫ ≤ ∥x∥ * ∥y∥ := by linarith,
+    have h₄ : re ⟪y, x⟫ ≤ ∥x∥ * ∥y∥ := by rwa [←inner_conj_sym, conj_re],
+    have : ∥x + y∥ * ∥x + y∥ ≤ (∥x∥ + ∥y∥) * (∥x∥ + ∥y∥),
+    { simp [←inner_self_eq_norm_square, inner_add_add_self, add_mul, mul_add, mul_comm],
+      linarith },
+    exact nonneg_le_nonneg_of_squares_le (add_nonneg (sqrt_nonneg _) (sqrt_nonneg _)) this
+  end,
+  norm_neg := λ x, by simp only [norm, inner_neg_left, neg_neg, inner_neg_right] }
+
+local attribute [instance] to_normed_group
+
+/-- Normed space structure constructed from a `inner_product_space.core` structure -/
+def to_normed_space : normed_space 𝕜 F :=
+{ norm_smul_le := assume r x,
+  begin
+    rw [norm_eq_sqrt_inner, inner_smul_left, inner_smul_right, ←mul_assoc],
+    rw [conj_mul_eq_norm_sq_left, of_real_mul_re, sqrt_mul, ←inner_norm_sq_eq_inner_self, of_real_re],
+    { simp [sqrt_norm_sq_eq_norm, is_R_or_C.sqrt_norm_sq_eq_norm] },
+    { exact norm_sq_nonneg r }
+  end }
+
+end inner_product_space.of_core
+
+/-- Given a `inner_product_space.core` structure on a space, one can use it to turn
+the space into an inner product space, constructing the norm out of the inner product -/
+def inner_product_space.of_core [add_comm_group F] [semimodule 𝕜 F]
+  (c : inner_product_space.core 𝕜 F) : inner_product_space 𝕜 F :=
+begin
+  letI : normed_group F := @inner_product_space.of_core.to_normed_group F 𝕜 _ _ _ c,
+  letI : normed_space 𝕜 F := @inner_product_space.of_core.to_normed_space F 𝕜 _ _ _ c,
+  exact { norm_sq_eq_inner := λ x,
+    begin
+      have h₁ : ∥x∥^2 = (sqrt (re (c.inner x x))) ^ 2 := rfl,
+      have h₂ : 0 ≤ re (c.inner x x) := inner_product_space.of_core.inner_self_nonneg,
+      simp [h₁, sqr_sqrt, h₂],
+    end,
+    ..c }
+end
+
+/-! ### Properties of inner product spaces -/
+
+variables {α β γ : Type*}
```

Comment:
In the upper part of the file, the notation for the vector field is `F`, which is good because it helps the reader. Could you use here `E`, `F` and `G` instead of weird greek letters?